### PR TITLE
feat: add tab components with active and hover state

### LINKED ISSUE #25: [COMPONENT] Tabs
We need tabs to be designed

- [x] Designs https://www.figma.com/file/aAZNLti1x7RHcKO2aaVdyh/Selleo-Design-System?node-id=2430%3A4658&t=kO4aJ2U5f5S3KHJz-0
- [x] Implementation

New tabs are designed so we can proceed with the implementation.
- Both dark & light mode are designed so should be implemented
- Each state should be shown in the docs (default & active)
- Icon edge case should be shown

Notes
- [Figma Design](https://www.figma.com/file/aAZNLti1x7RHcKO2aaVdyh/Selleo-Design-System?node-id=2430%3A4658&t=kO4aJ2U5f5S3KHJz-0)

### PR DESCRIPTION
**Pull request by DawidHaratyk**
closes #25

- Add Tab component,
- Handled states of a tab:
   - default
   - active
   - hover
   - dark mode

### CODE REVIEW

File: packages/selleo-design-core/src/Tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  active: boolean;
+  icon?: JSX.Element;
+};
+
+export function Tab({ active, icon }: TabProps) {
+  const classes = cx(
+    "px-2 py-1 text-base font-normal rounded-lg cursor-pointer",
+    {
+      "text-[#FF6D2A] bg-[#FFF8F5]": active,
+      "text-[#4C4D52] bg-white hover:bg-[#FFF8F5] dark:text-white dark:bg-[#2F3033] dark:hover:bg-[#4C4D52]":
```
Comment (k1eu) [inline packages/selleo-design-core/src/Tabs.tsx:14]: please use the colors that are initated in tailwind.config.cjs. They're from the design system core and should be the only one that we use

File: packages/selleo-design-core/src/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        !active,
+    }
+  );
+
+  return (
+    <span class={classes}>
+      {icon && "x "}
```
Comment (k1eu) [inline packages/selleo-design-core/src/Tabs.tsx:21]: please use the provided icon from props and then you can use in .astro component icons that we have in codebase already

File: apps/docs/src/pages/tabs.mdx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { Tab } from "@selleo/core/src/Tabs";
+
+import Preview from "../components/Preview.astro";
+import Description from "../components/Description.astro";
+
+<Description>
+  Below you can find a tabs component styled with tailwindcss.
+</Description>
+
+<Preview component={Tab} name="Default tab" active={false} />
```
Comment (k1eu) [inline apps/docs/src/pages/tabs.mdx:16]: default tab in dark mode doesn't have a bg on figma
image

File: apps/docs/src/pages/tabs.mdx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+<Preview component={Tab} name="Active tab" active={true} />
+
+<Preview
+  component={Tab}
+  name="Tab with icon"
+  active={false}
+  icon={<span>x</span>}
+/>
```
Comment (k1eu) [inline apps/docs/src/pages/tabs.mdx:25]: please add a row of tabs view like on the top of the design
multiple tabs with one of them selected

**Review by k1eu — CHANGES_REQUESTED** [review]

File: apps/docs/src/pages/tabs.mdx (hunk 3 above)
Comment (michalgren) [reply apps/docs/src/pages/tabs.mdx:16]: Default tab shouldn't have bg on both modes

File: packages/selleo-design-core/src/Tabs.tsx (hunk 1 above)
Comment (DawidHaratyk) [reply packages/selleo-design-core/src/Tabs.tsx:14]: Yes, you'are right

File: apps/docs/src/pages/tabs.mdx (hunk 4 above)
Comment (DawidHaratyk) [reply apps/docs/src/pages/tabs.mdx:25]: Done

File: apps/docs/src/pages/tabs.mdx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+<Preview component={Tab} name="Default tab" active={false} />
+
+<Preview component={Tab} name="Active tab" active={true} />
+
+<Preview
+  component={Tab}
+  name="Tab with icon"
+  active={false}
+  icon={<span>x</span>}
```
Comment (k1eu) [inline apps/docs/src/pages/tabs.mdx:24]: there are icons in design-system-core tha you can use here

**Review by k1eu — CHANGES_REQUESTED** [review]

File: packages/selleo-design-core/src/Tabs.tsx (hunk 2 above)
Comment (DawidHaratyk) [reply packages/selleo-design-core/src/Tabs.tsx:21]: Okay, done

File: apps/docs/src/pages/tabs.mdx (hunk 5 above)
Comment (DawidHaratyk) [reply apps/docs/src/pages/tabs.mdx:24]: Yes, done

**Review by k1eu — APPROVED** [review]

**Comment by k1eu** [comment]
@DawidHaratyk pls resolve conflict and merge

**Review by michalgren — APPROVED** [review]

**Pull request merged**